### ISSUE TITLE
Unable to select message response, quoting issue?

### ISSUE THREAD

**Issue by raiguard** [kakoune-lsp/kakoune-lsp#725]
LuaLS uses `window/showMessageRequest` to verify that the user wants to run a third-party plugin. However, when I select either option, it does not work.

![image](https://github.com/kakoune-lsp/kakoune-lsp/assets/3515394/d88c7872-db01-4131-850d-914d315dedd5)

![image](https://github.com/kakoune-lsp/kakoune-lsp/assets/3515394/48db38ea-6669-4e66-901a-110dd938d322)
 
I'm using LuaLS 3.7.3. This worked in the past so something regressed recently. I can perform a bisect if desired.

**Comment by krobelus** [comment]
Probably f42ebfa (Temporarily vendor new Kakoune menu, 2024-01-10).
Can you either provide a debug log or easy steps to reproduce? I'm trying to insert a single quote and Kakoune blocks completely :/

**Comment by krobelus** [comment]
nevermind I figured it out

**Comment by krobelus** [comment]
nope I didn't. Can you give debug log, simply run

```
HOME=$(mktemp -d) kak -e '
    eval %sh{kak-lsp --kakoune -s $kak_session}
    set global lsp_cmd "kak-lsp -s %val{session} -vvvv --log /tmp/kak-lsp.log"
    lsp-enable'
```

then open a lua file to reproduce and post `/tmp/kak-lsp.log`

(we should make this more ergonomic)

**Comment by raiguard** [comment]
[kak-lsp.log](https://github.com/kakoune-lsp/kakoune-lsp/files/14365179/kak-lsp.log)

**Comment by krobelus** [comment]
this should do it

```diff
diff --git a/rc/lsp.kak b/rc/lsp.kak
index b08cec1..114132c 100644
--- a/rc/lsp.kak
+++ b/rc/lsp.kak
@@ -1930,7 +1930,11 @@ define-command -hidden lsp-show-message-request -params 4.. -docstring %{
         evaluate-commands %sh{
             on_abort="$2"
             shift 2
-            echo lsp-menu -on-abort "$on_abort" "$@"
+            printf "lsp-menu -on-abort %s" "$on_abort"
+            for arg
+            do
+                printf " '%s'" "$(printf %s "$arg" | sed "s/'/''/g")"
+            done
         }
     }
 }
```

**Comment by raiguard** [comment]
Unfortunately it's still not working:
![image](https://github.com/kakoune-lsp/kakoune-lsp/assets/3515394/5db034f1-cea4-4322-8af9-655d0b7b3309)

![image](https://github.com/kakoune-lsp/kakoune-lsp/assets/3515394/cdbf8468-30f7-46b2-bc97-ec548080a2d5)

Debug log:
[kak-lsp.log](https://github.com/kakoune-lsp/kakoune-lsp/files/14365953/kak-lsp.log)

**Comment by krobelus** [comment]
sorry should be fixed now by 516cba9 (Truncate window/showMessageRequest titles at the first line break, 2024-02-22), thanks for reporting! I don't think this is a regression in kakoune-lsp but I didn't check.

With the log it was very easy to reproduce, I only had to `eval` the `lsp-show-message-request` command
